evalml/pipelines/utils.py

```diff
@@ -267,5 +270,5 @@ def _make_component_list_from_actions(actions):
     components = []
     for action in actions:
         if action.action_code == DataCheckActionCode.DROP_COL:
-            components.append(DropColumns(columns=action.details["columns"]))
+            components.append((DropColumns, {"columns": action.details["columns"]}))
```

Comment:
@angela97lin I think there's a mismatch. This code is expecting `action.details["columns"]` but our data checks populate `details={"column": col_name}`, e.g. the id column data check, no variance data check. I think we should make them all be "columns" cause that's what the drop columns expects.

Reply:
Agree here! Also, would it be possible to also condense these same actions into a singular component? For instance, if `_make_component_list_from_actions` returns a list of 5 `DropColumns` tuples, each referring to a different column, I'd prefer to condense the columns we need to drop into a list and pass that to a single `DropColumns` component. This would keep the component graph cleaner for the user as well.

Reply:
Oopsies, thank you for catching this!

@bchen1116 and @freddyaboulton, thank you both for commenting RE: consolidating columns into one component. I think the initial idea was that it would be easier to see if there were multiple components, as opposed to combining a bunch of seemingly random columns together (ex: one ID col, one highly null col), and to map an action to a component. But I hear both of ya, it could be clearer if we just combined everything... Would love to discuss a bit post-standup before making the change heh

Reply:
@angela97lin Sounds good! My main concern with having one component for each action is that our linear pipeline graph doesn't support it. To your point, I don't think combining all actions into a single component is the only solution. I think making sure they each have unique ways is also a possible solution. We could also consider a hybrid approach where we group all the drop columns per data check, e.g. the IdColumnDatacheck produces one component for dropping all id columns called "Drop Id Columns."